pkg/otlp/metrics/ttlcache.go

```diff
@@ -82,6 +82,13 @@ func (t *ttlCache) putAndGetDiff(
 			return 0, false
 		}
 		dx = val - cnt.value
+
+		// For the purpose of the POC using a map here for simplicity. Will likely 
+		// need to create a new func to use in this case.
+		if ok := rateAsGaugeMetrics[dimensions.name]; ok == "ok" {
+			dx = dx / time.Duration(ts - cnt.ts).Seconds()
```

Comment:
(For the refactor) We should check that `dt` is not zero, I think it makes sense to just change the `if cnt.ts > ts` above

Reply:
> We should check that dt is not zero

Not sure to understand the suggestion, what is `dt` ?

Reply:
Sorry, I meant the difference between `ts` and `cnt.ts`, the rate is the ratio between the delta in values and the delta in time, which you usually represent as $\mathrm{d}t$: 

$$\mathrm{rate} = \frac{\mathrm{d}x}{\mathrm{d}t} = \frac{x_{\mathrm{now}} - x_{\mathrm{prev}}}{t_{\mathrm{now}} - t_{\mathrm{prev}}}$$

Reply:
Understood, thanks for clarifying 👍